### ISSUE TITLE
在使用pipeline中的HomoLR模块时发现输出只有model没有data

### ISSUE THREAD

**Issue by oceanqdu** [FederatedAI/FATE#4366]
在使用pipeline进行横向逻辑回归时发现Evaluation模块并没有起效，进入borad发现HomoLR模块的输出只有model没有data
![image](https://user-images.githubusercontent.com/49178688/194743854-3f93ac58-8a57-40ce-8504-c63eae107a29.png)
因此Evaluation模块没有得到输入才没有任何输出。
![image](https://user-images.githubusercontent.com/49178688/194743891-ba49ae6f-30d4-46c6-9f93-d8831c860f0c.png)
在我使用dsl提交任务时则没有这个问题。
在使用pipeline的HomoLR模块时需要为该模块指定data输出吗？还是需要加什么参数进行修改？
我的代码如下：
`from pipeline.backend.pipeline import PipeLine
from pipeline.component import Reader,DataTransform,HomoLR,Evaluation,FeatureScale
from pipeline.interface import Data
import json
pipeline=PipeLine().set_initiator(role='guest',party_id=9999).set_roles(guest=9999,host=10000,arbiter=9999)

reader_0=Reader(name="reader_0")

reader_0.get_party_instance(role='guest',party_id=9999).component_param(
    table={"name":"bank_9999","namespace":"experiment"}
)

reader_0.get_party_instance(role='host',party_id=10000).component_param(
    table={"name":"bank_10000","namespace":"experiment"}
)

datatransform_0=DataTransform(name="datatransform_0",with_label=True,output_format="dense")

datatransform_0.get_party_instance(role='guest',party_id=9999)
datatransform_0.get_party_instance(role='host',party_id=10000)

scale_0=FeatureScale(name="scale_0")

param = {
        "penalty": "L2",
        "optimizer": "sgd",
        "tol": 1e-05,
        "alpha": 0.01,
        "max_iter": 30,
        "early_stop": "diff",
        "batch_size": -1,
        "learning_rate": 0.15,
        "decay": 1,
        "decay_sqrt": True,
        "init_param": {
            "init_method": "zeros"
        },
        "encrypt_param": {
            "method": None
        },
        "cv_param": {
            "n_splits": 4,
            "shuffle": True,
            "random_seed": 33,
            "need_cv": False
        }
}

homo_lr_0 = HomoLR(name='homo_lr_0',**param)

evaluation_0=Evaluation(name="evaluation_0",eval_type="binary")

#添加模块

pipeline.add_component(reader_0)
pipeline.add_component(datatransform_0,data=Data(data=reader_0.output.data))
pipeline.add_component(scale_0,data=Data(data=datatransform_0.output.data))
pipeline.add_component(homo_lr_0,data=Data(data=scale_0.output.data))
pipeline.add_component(evaluation_0,data=Data(data=homo_lr_0.output.data))

pipeline.compile()

pipeline.fit()

pipeline.dump("model/mnist111_model.pkl")

print(json.dumps(pipeline.get_component("homo_lr_0").get_summary(), indent=4, ensure_ascii=False))
print('................................................................')
print(json.dumps(pipeline.get_component("evaluation_0").get_summary(), indent=4, ensure_ascii=False))
print('................................................................')
print(pipeline.get_component("homo_lr_0").get_output_data(limits=10))`
运行成功，但是Broad中的Evaluation为空

**Comment by oceanqdu** [comment]
发现问题所在了
pipeline.add_component(homo_lr_0,data=Data(data=scale_0.output.data))
应该改为
pipeline.add_component(homo_lr_0,data=Data(train_data=scale_0.output.data))
